### PR TITLE
Added cra-template-imodeljs.md

### PR DESCRIPTION
**Pull request by ashish-srivastava-dev**
Added initial documentation for cra template imodeljs.

### CODE REVIEW

**Review by pmconne — APPROVED** [review]

File: docs/learning/tutorials/cra-template-imodeljs.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+## Developing a create-react-app with iModeljs template
```
Comment (calebmshafer) [inline docs/learning/tutorials/cra-template-imodeljs.md:1]: nit: iModeljs -> iModel.js

File: docs/learning/tutorials/cra-template-imodeljs.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+### Setup
+
+- [Install necessary prerequisites]($docs/getting-started/development-prerequisites)
+- Have access to a cloud hosted iModel. If you do not have access to one, follow one of our tutorials to [create an online iModel]($docs/learning/tutorials/index.md)
+- To use this template, add `--template @bentley/cra-template-imodeljs` when creating a new app. You should also use the @bentley/react-scripts scripts version to compile your application.
```
Comment (calebmshafer) [inline docs/learning/tutorials/cra-template-imodeljs.md:7]: Can we update the wording here to make it similar to the explicit instructions in #307 where it walks through how to use react-scripts partially as well? Hopefully that will make it clearer how to use it for someone new to CRA templates.

File: docs/learning/tutorials/cra-template-imodeljs.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+For example for creating a React app with iModeljs template, use the following command from command prompt.
```
Comment (calebmshafer) [inline docs/learning/tutorials/cra-template-imodeljs.md:9]: nit: iModeljs -> iModel.js

**Review by calebmshafer — COMMENTED** [review]
In addition to the comments below, we should figure out where the entry point is for this tutorial.  It should be shown somewhere on the index.md page, `docs/learning/tutorials/index.md`.

**Review by calebmshafer — CHANGES_REQUESTED** [review]
explicitly marking as request changes to stop the auto-merging.

File: docs/learning/tutorials/cra-template-imodeljs.md (hunk 2 above)
Comment (ashish-srivastava-dev) [reply docs/learning/tutorials/cra-template-imodeljs.md]: Modified it, but I need to give links for the npm package and the git repo (for cra-template-iModeljs) when [that pull request](https://github.com/imodeljs/create-react-app/pull/25) is completed.

Comment (ashish-srivastava-dev) [reply docs/learning/tutorials/cra-template-imodeljs.md]: Let me know if any more changes are needed.

File: docs/learning/tutorials/cra-template-imodeljs.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+## Developing a create-react-app with iModel.js template
+
```
Comment (williamkbentley) [inline docs/learning/tutorials/cra-template-imodeljs.md:2]: Can you add some context after header and before ### Setup. If someone stumbles across this page they wouldn't know what "Developing a create-react-app with iModel.js template" would be used for. I think the first 8 lines in the README would be good

File: docs/learning/tutorials/cra-template-imodeljs.md (hunk 2 above)
Comment (williamkbentley) [reply docs/learning/tutorials/cra-template-imodeljs.md]: @calebmshafer we will address reorganizing the tutorials page in a subsequent PR.

File: docs/learning/tutorials/cra-template-imodeljs.md (hunk 4 above)
Comment (ashish-srivastava-dev) [reply docs/learning/tutorials/cra-template-imodeljs.md:2]: Done.

**Comment by calebmshafer** [comment]
Closing this out due to us moving to the itwin viewer CRA template as a default.  If in the future we want to circle back to a non-itwin platform based CRA template, the template itself would need some work before it could get released.

**Pull request closed without merging**